src/api/domain/metadata-refreshes-api.ts

```diff
@@ -0,0 +1,629 @@
+     * @memberof MetadataRefreshesApiGetMetadataRefreshErrors
+     */
+    readonly xImxEthTimestamp: string
+
+    /**
+     * Wallet Address that signed the signature
+     * @type {string}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshErrors
+     */
+    readonly xImxEthAddress: string
+
+    /**
+     * Page size of the result
+     * @type {number}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshErrors
+     */
+    readonly pageSize?: number
+
+    /**
+     * Cursor
+     * @type {string}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshErrors
+     */
+    readonly cursor?: string
+}
+
+/**
+ * Request parameters for getMetadataRefreshResults operation in MetadataRefreshesApi.
+ * @export
+ * @interface MetadataRefreshesApiGetMetadataRefreshResultsRequest
+ */
+export interface MetadataRefreshesApiGetMetadataRefreshResultsRequest {
+    /**
+     * The metadata refresh ID
+     * @type {string}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshResults
+     */
+    readonly refreshId: string
+
+    /**
+     * String created by signing wallet address and timestamp. See https://docs.x.immutable.com/docs/generate-imx-signature
+     * @type {string}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshResults
+     */
+    readonly xImxEthSignature: string
+
+    /**
+     * Unix Epoc timestamp
+     * @type {string}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshResults
+     */
+    readonly xImxEthTimestamp: string
+
+    /**
+     * Wallet Address that signed the signature
+     * @type {string}
+     * @memberof MetadataRefreshesApiGetMetadataRefreshResults
+     */
+    readonly xImxEthAddress: string
+}
+
+/**
+ * Request parameters for requestAMetadataRefresh operation in MetadataRefreshesApi.
+ * @export
+ * @interface MetadataRefreshesApiRequestAMetadataRefreshRequest
+ */
+export interface MetadataRefreshesApiRequestAMetadataRefreshRequest {
+    /**
+     * String created by signing wallet address and timestamp. See https://docs.x.immutable.com/docs/generate-imx-signature
+     * @type {string}
+     * @memberof MetadataRefreshesApiRequestAMetadataRefresh
+     */
+    readonly xImxEthSignature: string
+
+    /**
+     * Unix Epoc timestamp
+     * @type {string}
+     * @memberof MetadataRefreshesApiRequestAMetadataRefresh
+     */
+    readonly xImxEthTimestamp: string
+
+    /**
+     * Wallet Address that signed the signature
+     * @type {string}
+     * @memberof MetadataRefreshesApiRequestAMetadataRefresh
+     */
+    readonly xImxEthAddress: string
+
+    /**
+     * Create metadata refresh request
+     * @type {CreateMetadataRefreshRequest}
+     * @memberof MetadataRefreshesApiRequestAMetadataRefresh
+     */
+    readonly createMetadataRefreshRequest: CreateMetadataRefreshRequest
+}
+
+/**
+ * MetadataRefreshesApi - object-oriented interface
+ * @export
+ * @class MetadataRefreshesApi
+ * @extends {BaseAPI}
+ */
+export class MetadataRefreshesApi extends BaseAPI {
```

Comment:
Not a blocker for this PR but we will need these methods from the new MetadataRefreshesApi to ImmutableX class.

Reply:
Yup. Can be done separately or even post 1.0.